File: EvaluacionTecnicaBackend/Inventario.Productos.Api/Modelos/Producto.cs
namespace Inventario.Productos.Api.Modelos;

public class Producto
{
    public Guid Id { get; set; }
    public string Nombre { get; set; } = string.Empty;
    public string? Descripcion { get; set; }
    public string Categoria { get; set; } = string.Empty;
    public string? Imagen { get; set; }
    public decimal Precio { get; set; }
    public int Stock { get; set; }
    public bool Activo { get; set; } = true;
    public DateTime FechaCreacion { get; set; } = DateTime.UtcNow;
    public DateTime? FechaActualizacion { get; set; }
}

File: EvaluacionTecnicaFront/src/app/core/models/producto.ts
export interface Producto {
  id: string;
  nombre: string;
  descripcion?: string | null;
  categoria: string;
  imagen?: string | null;
  precio: number;
  stock: number;
  activo: boolean;
  fechaCreacion: string;
  fechaActualizacion?: string | null;
}

export interface CrearProducto {
  nombre: string;
  descripcion?: string | null;
  categoria: string;
  imagen?: string | null;
  precio: number;
  stock: number;
}

export interface ActualizarProducto {
  nombre: string;
  descripcion?: string | null;
  categoria: string;
  imagen?: string | null;
  precio: number;
}

export interface FiltrosProductos {
  pagina: number;
  tamanoPagina: number;
  nombre?: string | null;
  categoria?: string | null;
  precioMinimo?: number | null;
  precioMaximo?: number | null;
  stockMinimo?: number | null;
  stockMaximo?: number | null;
}
